packages/flutter/lib/src/rendering/editable.dart

```diff
@@ -1790,42 +1797,64 @@ class RenderEditable extends RenderBox with RelayoutWhenSystemFontsChangeMixin,
   /// This does not require the layout to be updated.
   double get preferredLineHeight => _textPainter.preferredLineHeight;
 
-  double _preferredHeight(double width) {
-    // Lock height to maxLines if needed.
-    final bool lockedMax = maxLines != null && minLines == null;
-    final bool lockedBoth = minLines != null && minLines == maxLines;
-    final bool singleLine = maxLines == 1;
-    if (singleLine || lockedMax || lockedBoth) {
-      return preferredLineHeight * maxLines!;
-    }
-
-    // Clamp height to minLines or maxLines if needed.
-    final bool minLimited = minLines != null && minLines! > 1;
-    final bool maxLimited = maxLines != null;
-    if (minLimited || maxLimited) {
-      _layoutText(maxWidth: width);
-      if (minLimited && _textPainter.height < preferredLineHeight * minLines!) {
-        return preferredLineHeight * minLines!;
-      }
-      if (maxLimited && _textPainter.height > preferredLineHeight * maxLines!) {
-        return preferredLineHeight * maxLines!;
-      }
+  int? _cachedLineBreakCount;
+  int _countHardLineBreaks(String text) {
+    final int? cachedValue = _cachedLineBreakCount;
+    if (cachedValue != null) {
+      return cachedValue;
+    }
+    int count = 0;
+    for (int index = 0; index < text.length; index += 1) {
+      // TODO(LongCatIsLooong): account for other line terminators, or get it
+      // from ui.Paragraph.
+      count += text.codeUnitAt(index) == 0x0A ? 1 : 0;
```

Comment:
Nit: Maybe make 0x0A a constant for clarity about what it is, depending on your TODO above.